### PR TITLE
Fixed react table's display value formatting and added test cases

### PR DESCRIPTION
**Pull request by twheys**
Found some issues with display formatted, hard coded values from testing and some cases not covered, such as adding commas to numeric values.

### CODE REVIEW

File: fireant/slicer/widgets/reacttable.py [hunk 1]
```diff
@@ -69,25 +69,20 @@ def fillna_index(index, value):
...
-          suffix=(item.suffix or ""),
-          value=value if item.precision is None else '{:.{precision}f}'.format(0.1234567890, precision=2)
-    )
+def display_value(value, prefix=None, suffix=None, precision=None):
+    if isinstance(value, bool):
+        value = str(value).lower()
 
+    formatted_value = ('{:.{precision}f}'.format(value, precision=precision)
+                       if precision is not None and isinstance(value, float) else
+                       '{:,}'.format(value)
```
Comment (mikeengland) [inline fireant/slicer/widgets/reacttable.py]: Is it possible to split this? This is really hard to follow.

Comment (twheys) [reply fireant/slicer/widgets/reacttable.py]: would be nice if python supported switch statements. Does it help to follow it if you think of it as `condition 1 -> outcome 1 ELSE condition 2 -> outcome 2 ELSE outcome 3`?

Comment (twheys) [reply fireant/slicer/widgets/reacttable.py]: i broke it up to make it easier to read

Comment (mikeengland) [reply fireant/slicer/widgets/reacttable.py]: I personally think it would be a lot easier to read just with separated if statements.
